### DISCUSSION TITLE
Extract image data from pdf

### DISCUSSION THREAD

**Discussion by shubhamk-2610** [pymupdf/PyMuPDF#3475] [Looking for help]
I need to get the coordinates and the alt text of all the images present in the pdf.

Thank you.

**Comment by JorjMcKie** [comment]
Getting an image's coordinates on the page is no problem: please consult [get_image_rects](https://pymupdf.readthedocs.io/en/latest/page.html#Page.get_image_rects).
A PDF does not store things like Alt text. It is just text. You in your program must find a way to locate any text in an image's vicinity and try to understand if it may be meant to be Alt text.

**Reply by shubhamk-2610** [reply]
I have multiple same images on the pdf and the only afaik to differentiate them is the alt text
I think the alt text is present in the pdf tree structure. Can we extract it from there and then map it to the coordinates of the image somehow?

**Reply by JorjMcKie** [reply]
Sorry, there is no way to do that.
